src/firmware/shared/math/matrix.h

```diff
@@ -1,73 +1,85 @@
+#pragma once
 
-
+#include <stdlib.h>
 /**
  * A representation of a matrix that has an arbitrary number of rows and columns specified
  * by n_cols and n_rows. The recommended way to use this struct is to use the get_matrix
  * function, which will allocate the memory required for each of the rows in the matrix
  * and each of the entries in each of those rows. It will also assign the n_cols and
- * n_rows for the matrix. Then when you are done with the matrix, you will need to call
- * free_matrix.
+ * n_rows for the matrix.
+ *
+ * *******************************************************************
+ * NOTE: Then when you are done with the matrix, you will need to call
+ * shared_matrix_destroy().
+ * ********************************************************************
  */
-typedef struct
-{
-    float **rows;
-    int n_cols;
-    int n_rows;
-} Matrix;
+typedef struct Matrix Matrix_t;
 
 /**
  * Gets a matrix of the given size. This uses dynamic memory allocation so make sure to
- * call free_matrix when you are done using it if you don't want memory leaks.
+ * call shared_matrix_destroy when you are done using it if you don't want memory leaks.
```

Comment:
So to be consistent with other pseudo-classes, we don't specify in the constructor that the destructor needs to be called. We could consider adding it to every constructor, but that seems tricky to enforce.

``` suggestion
 * Create a matrix with the given number of rows and columns
```

Reply:
I've opened https://github.com/UBC-Thunderbots/Software/issues/1627 which should be a more rigorous and automatic manner of checking for memory leaks.

Reply:
Added